pallets/collator-selection/src/weights.rs

```diff
@@ -36,45 +36,44 @@ pub trait WeightInfo {
 pub struct SubstrateWeight<T>(PhantomData<T>);
 impl<T: frame_system::Config> WeightInfo for SubstrateWeight<T> {
 	fn set_invulnerables(b: u32, ) -> Weight {
-		(28_060_000 as Weight)
-			// Standard Error: 1_000
-			.saturating_add((118_000 as Weight).saturating_mul(b as Weight))
+		(18_563_000 as Weight)
+			// Standard Error: 0
+			.saturating_add((68_000 as Weight).saturating_mul(b as Weight))
 			.saturating_add(T::DbWeight::get().writes(1 as Weight))
 	}
 	fn set_desired_candidates() -> Weight {
-		(25_000_000 as Weight)
+		(16_363_000 as Weight)
 			.saturating_add(T::DbWeight::get().writes(1 as Weight))
 	}
 	fn set_candidacy_bond() -> Weight {
-		(25_000_000 as Weight)
+		(16_840_000 as Weight)
 			.saturating_add(T::DbWeight::get().writes(1 as Weight))
 	}
 	fn register_as_candidate(c: u32, ) -> Weight {
-		(82_496_000 as Weight)
-			// Standard Error: 1_000
-			.saturating_add((266_000 as Weight).saturating_mul(c as Weight))
-			.saturating_add(T::DbWeight::get().reads(3 as Weight))
-			.saturating_add(T::DbWeight::get().writes(1 as Weight))
+		(71_196_000 as Weight)
+			// Standard Error: 0
+			.saturating_add((198_000 as Weight).saturating_mul(c as Weight))
+			.saturating_add(T::DbWeight::get().reads(4 as Weight))
+			.saturating_add(T::DbWeight::get().writes(2 as Weight))
 	}
 	fn leave_intent(c: u32, ) -> Weight {
-		(65_836_000 as Weight)
-			// Standard Error: 2_000
-			.saturating_add((273_000 as Weight).saturating_mul(c as Weight))
+		(55_336_000 as Weight)
+			// Standard Error: 0
+			.saturating_add((151_000 as Weight).saturating_mul(c as Weight))
 			.saturating_add(T::DbWeight::get().reads(1 as Weight))
-			.saturating_add(T::DbWeight::get().writes(1 as Weight))
+			.saturating_add(T::DbWeight::get().writes(2 as Weight))
 	}
 	fn note_author() -> Weight {
-		(106_000_000 as Weight)
+		(71_461_000 as Weight)
 			.saturating_add(T::DbWeight::get().reads(3 as Weight))
 			.saturating_add(T::DbWeight::get().writes(4 as Weight))
 	}
-	fn new_session(r: u32, c: u32, ) -> Weight {
-		(50_005_000 as Weight)
-			// Standard Error: 2_000
-			.saturating_add((8_000 as Weight).saturating_mul(r as Weight))
-			// Standard Error: 2_000
-			.saturating_add((291_000 as Weight).saturating_mul(c as Weight))
+	fn new_session(_r: u32, c: u32, ) -> Weight {
+		(37_164_000 as Weight)
+			// Standard Error: 5_000
+			.saturating_add((5_724_000 as Weight).saturating_mul(c as Weight))
 			.saturating_add(T::DbWeight::get().reads(3 as Weight))
+			.saturating_add(T::DbWeight::get().reads((1 as Weight).saturating_mul(c as Weight)))
```

Comment:
`c` is the number of collators?

Reply:
yup C is collators and R is collators to be removed